geoh5py/objects/object_base.py

```diff
@@ -492,8 +479,24 @@ def remove_children(self, children: list[Entity] | list[PropertyGroup]):
             from the workspace by
             :func:`~geoh5py.shared.weakref_utils.remove_none_referents`.
         """
-        super().remove_children(children)
-        self._property_groups = None
+        if not isinstance(children, list):
+            children = [children]
+
+        for child in children:
+            if child not in self._children:
+                continue
+
+            self._children.remove(child)
+
+            if not self._property_groups:
+                continue
+
+            if isinstance(child, PropertyGroup):
+                self._property_groups.remove(child)
+            elif isinstance(child, Data):
```

Comment:
Are data always associated to property groups?
What if a Data is a 'raw' children of an object, we just want to remove it aren't we?

Reply:
The plain removal from the list happens above, then from the file below.